src/Symfony/Component/Validator/Constraints/ChoiceValidator.php

```diff
@@ -55,6 +55,12 @@ public function validate($value, Constraint $constraint)
                 throw new ConstraintDefinitionException('The Choice constraint expects a valid callback.');
             }
             $choices = $choices();
+            if (!is_array($choices)) {
```

Comment:
I checked what we document for the `callback` option and we indeed do not document anything else than returning an array. We should therefore drop the `iterator_to_array()` part as adding that would be a new feature. 

The aded code can then be simplified like this:

```php
if (!is_array($choices)) {
    throw new ConstraintDefinitionException(sprintf('The Choice constraint callback "%s" is expected to return an array, but returned "%s".', trim($this->formatValue($constraint->callback), '"'), get_debug_type($choices)));
}
```

Reply:
ok makes sense, I removed the `iterator_to_array()` part and the test associated.

Reply:
what is documented exactly?

https://github.com/symfony/symfony/blob/e59f035c0a389a83f6ac7fe3325a0e5cb6c18071/src/Symfony/Component/Validator/Constraints/Choice.php#L54

Reply:
oh the in_array check would fail for iterables